ribs/archives/_archive_base.py

```diff
@@ -569,18 +655,78 @@ def add_single(self, solution, objective, measures, metadata=None):
             the add operation. Refer to :meth:`add` for the meaning of the
             status and value.
         """
+        self._state["add"] += 1
+
         solution = np.asarray(solution)
-        objective = np.asarray(objective, dtype=self.dtype)  # 0-dim array.
+        objective = self.dtype(objective)
         measures = np.asarray(measures)
+        index = self.index_of_single(measures)
 
         check_1d_shape(solution, "solution", self.solution_dim, "solution_dim")
```

Comment:
These should be placed immediately after calling `asarray` on their respective arrays, so that we catch errors before anything else is done with the arrays.